pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,123 @@
-
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        client_api = gen_container_client()
-        cls.repositories_api = RepositoriesContainerApi(client_api)
-        cls.remotes_api = RemotesContainerApi(client_api)
-        cls.tags_api = ContentTagsApi(client_api)
-        cls.versions_api = RepositoriesContainerVersionsApi(client_api)
+    return _synced_repository_with_filtered_tags
 
-        cls.repository = None
 
-    def test_sync_with_non_existing_included_tag(self):
-        """Check whether the sync machinery ignores a non-existing tag."""
-        include_tags = ["manifest_a", "non_existing_manifest"]
-        self.sync_repository_with_filtered_tags(include_tags)
+@pytest.mark.parallel
+def test_basic_sync(
+    container_repo, container_remote, container_repository_api, container_tag_api, sync
+):
 
-        self.assert_synced_tags(["manifest_a"])
+    sync(container_repo, container_remote)
+    repository = container_repository_api.read(container_repo.pulp_href)
+    first_sync_tags_named_a = container_tag_api.list(name="manifest_a")
 
-    def test_sync_with_included_tags(self):
-        """Test whether the repository is synced only with included tags."""
-        include_tags = ["manifest_a", "manifest_b", "manifest_c"]
-        self.sync_repository_with_filtered_tags(include_tags)
+    assert "versions/1/" in repository.latest_version_href
 
-        self.assert_synced_tags(include_tags)
+    latest_version_href = repository.latest_version_href
+    sync(
+        container_repo, container_remote
+    )  # We expect that this second sync doesn't create a new repo version
+    repository = container_repository_api.read(repository.pulp_href)
+    second_sync_tags_named_a = container_tag_api.list(name="manifest_a")
 
-    def test_sync_with_included_tags_using_wildcard(self):
-        """Test whether the repository is synced only with included tags that use wildcards."""
-        include_tags = [
-            "ml_iv",
-            "ml_ii",
-            "manifest_a",
-            "manifest_b",
-            "manifest_c",
-            "manifest_d",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(["ml_??", "manifest*"])
-
-        self.assert_synced_tags(include_tags)
-
-    def test_sync_with_complex_filtering(self):
-        """Test sync repository with included and excluded that use wildcards."""
-        include_tags = [
-            "manifest_a",
-            "manifest_c",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(
-            include_tags=["manifest_*"], exclude_tags=["*_[bd]"]
-        )
+    assert repository.latest_version_href == latest_version_href
 
-        self.assert_synced_tags(include_tags)
+    # Ensure that sync does not create orphan tags
+    assert first_sync_tags_named_a.count == 1
+    assert second_sync_tags_named_a.count == 1
```

Comment:
Maybe, I would change the logic of the assertions as follows:
1. After the first sync, retrieve a list of all synced tags from `first_synced_tags = container_tag_api.list(repository_version=repository.latest_version_href).results`.
2. After the second sync, retrieve a list of synced tags again, like `second_sync_tags = container_tag_api.list(repository_version=repository.latest_version_href).results`
3. Finally, do the following assertion: `assert sorted(map(lambda tag: tag["pulp_href"], first_sync_tags)) == sorted(map(lambda tag: tag["pulp_href"], second_sync_tags)).
`

Right now, we test the presence only of one tag. With this, we will test all synced tags. Furthermore, your test can fail in the future because you are filtering tags without specifying the version of a repository. The number of globally accessible tags may vary across the tests once they run in parallel.

How does it sound?